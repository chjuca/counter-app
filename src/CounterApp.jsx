import PropTypes from 'prop-types';
import { useState } from 'react';

const CounterApp = ({value}) => {

    const [counter, setCounter] = useState(value)

    const handleAdd = () => {
        setCounter(counter + 1)
    }

    const handleSubtract = () => {
        setCounter(counter - 1)
    }

    const handleReset = () => {
        setCounter(value)
    }

    return <>
        <h1>CounterApp</h1>
        <p>{counter}</p>

        <button aria-label="btn-increment" onClick={handleAdd}> +1 </button>
        <button aria-label="btn-decrement" onClick={handleSubtract }> -1 </button>
        <button aria-label="btn-reset" onClick={handleReset}> Reset </button>
    </>
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
}

// CounterApp.defaultProps = {
//     value: 5
// }

export default CounterApp