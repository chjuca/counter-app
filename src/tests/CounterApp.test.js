import { fireEvent, render, screen } from "@testing-library/react"
import CounterApp from "../CounterApp"

describe("Test <CounterApp/>", () =>{

    const value = 100

    test('Should match snapshot', () => {
        const {container} = render(<CounterApp value = {value}/>)
        expect (container).toMatchSnapshot()
    })

    test('should show initial value 100', () => {
        render(<CounterApp value={value}/>)
        expect (screen.getByRole('paragraph').innerHTML).toContain('100')
    })

    test('should increment with button +1', () => {
        render(<CounterApp value={value}/>)
        fireEvent.click(screen.getByText('+1'))
        expect (screen.getByRole('paragraph').innerHTML).toContain('101')
    })

    test('should decrement with button -1', () => {
        render(<CounterApp value={value}/>)
        fireEvent.click(screen.getByText('-1'))
        expect (screen.getByRole('paragraph').innerHTML).toContain('99')
    })

    test('should reset', () => {
        render(<CounterApp value={value}/>)
        fireEvent.click(screen.getByText('-1'))
        fireEvent.click(screen.getByText('+1'))
        fireEvent.click(screen.getByText('+1'))
        fireEvent.click(screen.getByText('+1'))
        fireEvent.click(screen.getByText('+1'))
        fireEvent.click(screen.getByRole('button', {name: "btn-increment"}))
        fireEvent.click(screen.getByText('Reset'))
        expect (screen.getByRole('paragraph').innerHTML).toContain('100')
    })


})