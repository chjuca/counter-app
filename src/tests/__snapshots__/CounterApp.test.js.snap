// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <CounterApp/> Should match snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <p>
    100
  </p>
  <button
    aria-label="btn-increment"
  >
     +1 
  </button>
  <button
    aria-label="btn-decrement"
  >
     -1 
  </button>
  <button
    aria-label="btn-reset"
  >
     Reset 
  </button>
</div>
`;
